### PR TITLE
modulesync 5.3.0 + general cleanup + EL7/EL8/EL9 acceptance testing

### PR DESCRIPTION
**Pull request by jhoblitt**
* The `lookup()` for the `ccs_monit::hwraid` hiera key has been removed and replaced with the `has_dellperc` fact. 
* All other `lookup()`s have been converted to class parameters.
* The dependency on `$profile::ccs::facts::main_interface` has been removed.

### CODE REVIEW

File: files/monit_hwraid [hunk 1]
```diff
@@ -21,7 +21,7 @@ type -ap $raid >& /dev/null || {
     exit 1
 }
 
-stat=$($raid $args 2> /dev/null | grep $patt)
+stat=$($raid "$args" 2> /dev/null | grep $patt)
```
Comment (glennmorris) [inline files/monit_hwraid:24]: That change is not right, we want $args to be split on whitespace.
(In general I find shellcheck mainly just an exercise in pedantry.)

**Review by glennmorris — CHANGES_REQUESTED** [review]
Thanks for this!
I would have preferred to have the functional changes (/usr/local/bin/monit, use of standard facts, etc) be in a separate pull(s) from the cosmetic/boilerplate changes, but it's not worth splitting it now.
I spotted one mistake in the shellcheck pacification (see comment), but otherwise LGTM.

**Comment by jhoblitt** [comment]
@glennmorris I agree that this PR got unwieldy. The changes were largely driven by getting the acceptance tests to pass without having to pull in the control repo.

File: files/monit_hwraid (hunk 1 above)
Comment (jhoblitt) [reply files/monit_hwraid]: Copy. I've fixed.  I often find shellcheck useful and it at least imposes something of a consistent style across repos.

**Comment by jhoblitt** [comment]
Note that the acceptance test here is only making sure that puppet runs without error. It isn't doing anything to check functionality.

**Review by glennmorris — APPROVED** [review]

**Pull request merged**